### PR TITLE
Adds support for Tier and SlaPolicy to the Job interface

### PR DESCRIPTION
**Pull request by zircote**
This adds the following function to the Job interface
- `Tier(tier *string) Job`
- `SlaPolicy(policy *aurora.SlaPolicy) Job`
- Test coverage for each in the end to end tests.

### CODE REVIEW

File: realis_e2e_test.go [hunk 1]
```diff
@@ -692,3 +691,89 @@ func TestRealisClient_ForceExplicitTaskReconciliation(t *testing.T) {
...
+			"create_service_with_sla_percentage_policy_test",
+			aurora.SlaPolicy{PercentageSlaPolicy: &aurora.PercentageSlaPolicy{Percentage: 0.25, DurationSecs: 30}},
+		},
+		{
+			"create_service_with_sla_coordinator_policy_test",
+			aurora.SlaPolicy{CoordinatorSlaPolicy: &aurora.CoordinatorSlaPolicy{
+				CoordinatorUrl: "http://localhost/endpoint", StatusKey: "aurora_test"}},
+		},
+	}
+	//fmt.Print(tt.args)
```
Comment (rdelval) [inline realis_e2e_test.go]: Nit: Can we get rid of this commented out print statement?

Comment (zircote) [reply realis_e2e_test.go]: yes, totally missed that, thank you.

**Comment by zircote** [comment]
@rdelval so a couple things poped up with the `SlaPolicy`
- the Jobs have to be run with `20` shards to use Sla 
- with hte Sla tests the tests run great than 10 minutes (go test times out)

I have not pushed changes to address these yet to my branch; however I wanted to get your input regarding how you would prefer to approach this. What I have done (not pushed yet) is
- increase test `-timeout 15m` in both the `runMacTests.sh` and `.travis.yml`
- I have added `3` additional mesos agents to the `docker-compose.yml` to provide enough resources to the mesos cluster to support the `20` shards for the SlaTests

I am not comfortable pushing this up to the PR until you provide me your guidance regarding preferences.

Thank you.

**Comment by rdelval** [comment]
@zircote one thing you can do is lower the `UpdateSettings.minWaitInInstanceRunningMs` to something like 1000 (one second). That'll reduce the length of the update.

To reduce the number of Mesos agents back down to one, we can reduce the resources required for each task to something like `<CPU .05, RAM 4,  Disk 5>` that should be enough breathing room for 20 instances.

Happy to hear that scaling this compose file to multiple Mesos Agents works fine. It may prove useful in the future 😄

**Comment by zircote** [comment]
So ultimately the only way I could get the SlaPolicy test to work was by adding 3 mesos agents to the docker compose and reducing the resources for the task in `0.1, 2, 2`. I updated the settings for the job update to a 15 second wait yet the total test time is still cutting it close to the default `10m` test timeout. I opted to update that to `15m` in both the mac shell script AND the travis.yml. I am happy to hear back your thoughts and feedback on this. I am not terribly excite changing those files without a resounding ok for you ( @rdelval ) though.

**Comment by rdelval** [comment]
Let's do it! If that's the only way to get it to work then so be it. Let's commit it for now and leave a TODO somewhere to see if we can scale back to a single Agent somehow.

File: job.go [hunk 2]
```diff
@@ -315,3 +317,17 @@ func (j *AuroraJob) Container(container Container) Job {
 
 	return j
 }
+
+// Set the Tier for the Job.
+func (j *AuroraJob) Tier(tier *string) Job {
```
Comment (rdelval) [inline job.go]: Let's make this function take a concrete type as an argument and then assign the address of it to the `.Tier` field. That way we don't have to create a pointer to pass to this function.

**Comment by rdelval** [comment]
@zircote looks like the added tests for SLA are failing. Any chance you can look into this?
https://travis-ci.com/paypal/gorealis/builds/95476949#L653

**Comment by rdelval** [comment]
@zircote any chance you can get around to this this week? If you need help finishing this patch let me know!

**Comment by rdelval** [comment]
@zircote still interested in getting this PR merged?

**Comment by rdelval** [comment]
Rebased and fixed failed tests in #99 since @zircote  has not been replying to messages. Thanks for your contribution!

**Pull request closed without merging**